src/main/java/org/commcare/formplayer/services/RestoreFactory.java

```diff
@@ -292,6 +300,42 @@ private UserSqlSandbox restoreUser(boolean skipFixtures) throws SyncRestoreExcep
                 UserSqlSandbox sandbox = getSqlSandbox();
                 FormplayerTransactionParserFactory factory = new FormplayerTransactionParserFactory(sandbox, true);
                 InputStream restoreStream = getRestoreXml(skipFixtures);
+                if (!shouldPurge) {
+                    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```

Comment:
Don't think we need to do any additional parsing to determine location has changed or not. Instead we can monitor the # of locations in user sandbox before and after the restore has happened to determine it.

Reply:
ahh, that's true. Though, instead of number of locations I'll still be comparing the contents of `uUser.getProperty("commcare_location_ids")` since it's probably common to change from one location to another (so you're still assigned to just one location).